### PR TITLE
28.1dev

### PR DESCRIPTION
**Pull request by yutsevich**

### CODE REVIEW

File: src/main/java/com/internet/shop/controllers/LoginController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        req.getRequestDispatcher("/WEB-INF/views/login.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("psw");
+
+        try {
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/LoginController.java]:
```suggestion
        String password = req.getParameter("psw");
        try {
```

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User user = userService.findByLogin(login).orElseThrow(() ->
+                        new AuthenticationException("Incorrect username or password"));
+
+        if (user.getPassword().equals(password)) {
+            return user;
+        } else {
+            throw new AuthenticationException("Incorrect username or password");
+        }
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: There is no need to throw the same exception with the same message.
Let's place both checks into the if block.

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+@WebFilter(filterName = "Authentication filter",
+        urlPatterns = {"/*"})
+public class AuthenticationFilter implements Filter {
+    public static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
+    }
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]:
```suggestion
    public void init(FilterConfig filterConfig) throws ServletException {
    }
```

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            return;
+        }
+
+        Long userId = (Long) req.getSession().getAttribute("user_id");
+        if (userId == null || userService.get(userId) == null) {
+            resp.sendRedirect("/login");
+            return;
+        }
+
+        filterChain.doFilter(req, resp);
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]:
```suggestion
        HttpServletResponse resp = (HttpServletResponse) servletResponse;
        String url = req.getServletPath();
        if (url.equals("/login") || url.equals("/registration")) {
            filterChain.doFilter(req, resp);
            return;
        }
        Long userId = (Long) req.getSession().getAttribute("user_id");
        if (userId == null || userService.get(userId) == null) {
            resp.sendRedirect("/login");
            return;
        }
        filterChain.doFilter(req, resp);
```

Comment (dariiabutorina) [reply src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]: Use the same code writing style during the whole project.
You're using such blank line separation in this method only.

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @Override
+    public void destroy() {
+
+    }
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]:
```suggestion
    public void destroy() {
    }
```

File: src/main/webapp/WEB-INF/views/login.jsp [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+<h4 style="color: red">${errorMsg}</h4>
+
+<form method="post" action="${pageContext.request.contextPath}/login">
+    Enter your login please: <input type="text" name="login" required placeholder="login"> <br>
+    Enter your password please: <input type="password" name="psw" required placeholder="password"> <br>
+
+    <button type="submit">Login</button>
+</form>
+
+</body>
```
Comment (dariiabutorina) [inline src/main/webapp/WEB-INF/views/login.jsp]:
```suggestion
<body>
<h4 style="color: red">${errorMsg}</h4>
<form method="post" action="${pageContext.request.contextPath}/login">
    Enter your login please: <input type="text" name="login" required placeholder="login"> <br>
    Enter your password please: <input type="password" name="psw" required placeholder="password"> <br>
    <button type="submit">Login</button>
</form>
</body>
```

**Review by dariiabutorina — CHANGES_REQUESTED** [review]

File: src/main/java/com/internet/shop/controllers/LoginController.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("psw");
+        try {
+            User user = authService.login(login, password);
+            HttpSession session = req.getSession();
+            session.setAttribute("user_id",user.getId());
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/LoginController.java]: let's make `user_id` a constant

File: src/main/java/com/internet/shop/security/AuthenticationService.java [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package com.internet.shop.security;
+
+import com.internet.shop.exceptions.AuthenticationException;
+import com.internet.shop.model.User;
+
+public interface AuthenticationService {
+    User login(String login, String password)
+            throws AuthenticationException;
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/security/AuthenticationService.java]: this fits in one line

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+                         FilterChain filterChain)
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        String url = req.getServletPath();
+        if (url.equals("/login") || url.equals("/registration")) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        Long userId = (Long) req.getSession().getAttribute("user_id");
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]: let's make `user_id` a constant

File: src/main/webapp/WEB-INF/views/login.jsp [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+<html>
+<head>
+    <title>Login</title>
+</head>
+<body>
+<h4 style="color: red">${errorMsg}</h4>
+<form method="post" action="${pageContext.request.contextPath}/login">
+    Enter your login please: <input type="text" name="login" required placeholder="login"> <br>
+    Enter your password please: <input type="password" name="psw" required placeholder="password"> <br>
+    <button type="submit">Login</button>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/login.jsp]: let's also add the link to register

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**